pipelines/backend_tests.go

```diff
@@ -2,21 +2,49 @@ package pipelines
 
 import (
 	"context"
+	"fmt"
+	"log"
+	"strings"
 
 	"dagger.io/dagger"
 	"github.com/grafana/grafana-build/containers"
 )
 
+var IntegrationDatabases = []string{"sqlite", "mysql", "postgres"}
+
+// contains returns true if the string v is in the slice arr
+func contains(arr []string, v string) bool {
+	for _, str := range arr {
+		if str == v {
+			return true
+		}
+	}
+	return false
+}
+
 // GrafanabackendTests runs the Grafana backend test containers for short (unit) and integration tests.
 func GrafanaBackendTests(ctx context.Context, d *dagger.Client, args PipelineArgs) error {
+
+	db := args.Context.String("database")
+
+	// verify the database is one of the available choices
+	if !contains(IntegrationDatabases, db) {
```

Comment:
Is it perhaps possible to do that validation step *within* the flag itself? Perhaps in the `Apply` method?

Reply:
see comment above about the validation.